usaspending_api/search/v2/views/spending_by_category_views/spending_by_recipient.py

```diff
@@ -64,18 +63,35 @@ def _get_recipient_id(row: list) -> str:
         )
 
     def build_elasticsearch_result(self, response: dict) -> List[dict]:
+        # Get the codes
+        recipient_info_buckets = response.get("group_by_agg_key", {}).get("buckets", [])
+        recipient_hashes = [bucket.get("key").split("/")[0] for bucket in recipient_info_buckets if bucket.get("key")]
+
+        # Get the current recipient info
+        current_recipient_info = {}
+        recipient_info_query = RecipientLookup.objects.filter(recipient_hash__in=recipient_hashes).values(
+            "duns", "legal_business_name", "recipient_hash"
+        )
+        for recipient_info in recipient_info_query.all():
+            current_recipient_info[str(recipient_info["recipient_hash"])] = recipient_info
 
+        # Build out the results
         results = []
-        location_info_buckets = response.get("group_by_agg_key", {}).get("buckets", [])
-        for bucket in location_info_buckets:
-            recipient_info = json.loads(bucket.get("key"))
+        for bucket in recipient_info_buckets:
+            result_hash, result_level = tuple(bucket.get("key").split("/")) if bucket.get("key") else (None, None)
+            result_hash_with_level = f"{result_hash}-{result_level}" if (result_hash and result_level) else None
+            recipient_info = current_recipient_info.get(result_hash) or {}
 
             results.append(
                 {
                     "amount": int(bucket.get("sum_field", {"value": 0})["value"]) / Decimal("100"),
-                    "recipient_id": recipient_info["hash_with_level"] or None,
-                    "name": recipient_info["name"] or None,
-                    "code": recipient_info["duns"] or "Recipient not provided",
+                    "recipient_id": result_hash_with_level,
+                    "name": (
+                        recipient_info["legal_business_name"].upper()
```

Comment:
This should already be uppercased if you are pulling the name from the recipient_lookup table.